two/src/main.cpp

```diff
@@ -0,0 +1,176 @@
+#include <iostream>
+
+// 2
+int a = 123;
+
+void f(const int a = 5)
+{
+  std::cout << a * 2 << std::endl;
+}
+
+/*
+  this function invokes f(), which receives an integer, doubles and prints it out
+  it has a default value assigned for passing nothing to f() will print 10
+*/
+void Two()
+{
+  f(1);
+  f(a);
+  int b = 3;
+  f(b);
+  int a = 4;
+  f(a);
+  f();
+}
+// 2
+
+// 3.1
+/*
+  The main function invokes printNum() however the function implementation is below,
+  to fix move the implementation above main() OR declare printNum() in a header or above main()
+*/
+// 3.1
+
+// 3.2
+/*
+  printNum() needs to be defined to accept an integer... I don't see any other fixes
+*/
+// 3.2
+
+// 3.3
+/*
+  The usage in main() indicates that doubleNumber() needs to mutate the passed in value.
+  To fix switch doubleNumber() to receive a reference to an integer
```

Comment:
Very true!
As a matter of our style (which is also common style), when we pass by reference we pass by const reference.
https://youilabs.atlassian.net/wiki/pages/viewpage.action?pageId=524322#C++CodingStandard-Allparameterspassedbyreferencemustbelabeledconst

So instead pass by pointer to mutate.